pallets/gear/rpc/src/lib.rs

```diff
@@ -303,4 +319,52 @@ where
             )
         })
     }
+
+    fn read_state(
+        &self,
+        program_id: H256,
+        at: Option<<Block as BlockT>::Hash>,
+    ) -> RpcResult<Bytes> {
+        let at = BlockId::hash(at.unwrap_or_else(||
+            // If the block hash is not supplied assume the best block.
+            self.client.info().best_hash));
+
+        self.run_with_api_copy(|api| api.read_state(&at, program_id).map(|r| r.map(Bytes)))
+    }
+
+    fn read_state_using_wasm(
+        &self,
+        program_id: H256,
+        fn_name: Bytes,
+        wasm: Bytes,
+        argument: Option<Bytes>,
+        at: Option<<Block as BlockT>::Hash>,
+    ) -> RpcResult<Bytes> {
+        let at = BlockId::hash(at.unwrap_or_else(||
+            // If the block hash is not supplied assume the best block.
+            self.client.info().best_hash));
+
+        self.run_with_api_copy(|api| {
+            api.read_state_using_wasm(
+                &at,
+                program_id,
+                fn_name.to_vec(),
+                wasm.to_vec(),
+                argument.map(|v| v.to_vec()),
+            )
+            .map(|r| r.map(Bytes))
+        })
+    }
+
+    fn read_metahash(
+        &self,
+        program_id: H256,
+        at: Option<<Block as BlockT>::Hash>,
+    ) -> RpcResult<H256> {
+        let at = BlockId::hash(at.unwrap_or_else(||
+            // If the block hash is not supplied assume the best block.
+            self.client.info().best_hash));
+
+        self.run_with_api_copy(|api| api.read_metahash(&at, program_id))
```

Comment:
I'm not sure if these methods need to be run within copied context.

Reply:
Just for confidence that nothing will be changed in memory of program during metahash fn (we can't restrict its realisation)